estimagic/differentiation/differentiation.py

```diff
@@ -195,13 +196,10 @@ def _no_extrapolation_hessian(internal_func, params_value, method):
 
 
 def _create_internal_func(func, params, func_kwargs):
-    def internal_func(x):
-        p = params.copy(deep=True)
-        p["value"] = x
-
+    @x_to_params(params)
+    def internal_func(p):
         func_value = func(p, **func_kwargs)
-        if isinstance(func_value, (pd.DataFrame, pd.Series)):
-            func_value = func_value.to_numpy()
```

Comment:
I think this line should not have been deleted, but I proposed a change to `x_to_params` that would make it unnecessary.